worker/broker.go

```diff
@@ -0,0 +1,510 @@
+		block, realCount, _, err := db.chain.FetchBlockByCount(-1)
+		if err != nil {
+			log.Errorf("MQTT fetch block failed, databaseID: %v, err: %v", dbID, err)
+			return
+		}
+
+		for index, qat := range block.QueryTxs {
+			payload := BrokerPayload{
+				BlockID:        realCount,
+				BlockIndex:     index,
+				ClientID:       qat.Request.Header.NodeID,
+				ClientSequence: qat.Request.Header.SeqNo,
+				Events:         convertToMQTTQuery(qat.Request.Payload.Queries),
+			}
+			if blockIndex == bStart {
+				if index < iStart {
+					continue
+				}
+			} else if blockIndex == bEnd {
+				if index > iEnd {
+					// Success publish all replay
+					return
+				}
+			}
+			err = c.PublishDSN(MQTTReplay, qat.Request.Header.DatabaseID, payload, payload.ClientID)
+			if err != nil {
+				log.Errorf("MQTT publish replay api failed, databaseID: %v, payload: %v, err: %v", qat.Request.Header.DatabaseID, payload, err)
+				// Cancel further publish if any error
+				return
+			}
+		}
+	}
+}
+
+func (c *MQTTClient) processCreateEvent(event *SubscribeEvent) {
+	log.Debugf("Create API does not support yet.\n")
+}
+
+func (c *MQTTClient) updateBlockLoop() {
+	var dbBlockLastIndex sync.Map
+	for {
+		select {
+		case <-c.updateCtx.Done():
+			return
+		case <-time.After(conf.GConf.SQLChainPeriod):
+			// TODO
+			// make it unblock
+			c.dbms.dbMap.Range(func(_, rawDB interface{}) bool {
+				db := rawDB.(*Database)
+
+				// fetch newest block
+				block, realCount, _, err := db.chain.FetchBlockByCount(-1)
+				if err != nil {
+					log.Errorf("MQTT fetch block failed: databaseID: %v, err: %v", db.dbID, err)
+					return false
+				}
+
+				// find last published block index
+				var lastIndex int32
+				li, ok := dbBlockLastIndex.Load(db.dbID)
+				if !ok {
+					lastIndex = 0
+				} else {
+					lastIndex = li.(int32)
+				}
+				if lastIndex == realCount {
+					log.Infof("MQTT fetch newest block not changed, databaseID: %v", db.dbID)
+					return true
+				}
+
+				// publish every query of newest block
+				err = nil
+				for index, qat := range block.QueryTxs {
+					payload := BrokerPayload{
+						BlockID:        realCount,
+						BlockIndex:     index,
+						ClientID:       qat.Request.Header.NodeID,
+						ClientSequence: qat.Request.Header.SeqNo,
+						Events:         convertToMQTTQuery(qat.Request.Payload.Queries),
+					}
+					err = c.PublishDSN(MQTTNewest, qat.Request.Header.DatabaseID, payload, payload.ClientID)
+					if err != nil {
+						log.Errorf("MQTT publish newest api failed, databaseID: %v, payload: %v, err: %v", qat.Request.Header.DatabaseID, payload, err)
+					}
+				}
+				if err != nil {
+					return false
+				}
+
+				// update last published block index
+				dbBlockLastIndex.Store(db.dbID, realCount)
+				return true
+			})
+
+		}
+	}
+}
+
+func (c *MQTTClient) PublishDSN(apiName MQTTAPI, databaseID proto.DatabaseID, payload BrokerPayload, requestClient proto.NodeID) error {
```

Comment:
**[golint]** <sub>reported by [reviewdog](https://github.com/haya14busa/reviewdog) :dog:</sub>
exported method MQTTClient.PublishDSN should have comment or be unexported